File: AI.Core/OperatorsImplimentation/BasicOperators/One.cs
using AI.Core.ChannelsImplimentation;

namespace AI.Core.OperatorsImplimentation.BasicOperators
{
    public class One : Operator
    {
        /// <summary>
        /// Always 1
        /// </summary>
        public One(Entity thisEntity)
            : base(thisEntity)
        {}

        public override void Action()
        {
            ExitContacts.SetValue(1);
        }
    }
}

File: NativeCore/OperatorsImplimentation/BasicOperators/One.cpp
#include "One.h"

using namespace AI::Core::ChannelsImplimentation;
namespace AI
{
	namespace Core
	{
		namespace OperatorsImplimentation
		{
			namespace BasicOperators
			{

				One::One(Entity *thisEntity) : Operator(thisEntity)
				{
				}

				void One::Action()
				{
					ExitContacts->SetValue(1);
				}
			}
		}
	}
}
